### PR TITLE
fix(dspv2): label should be check on existance and value as well

### PR DESCRIPTION
**Pull request by zdtsw**
- when label has value "true" means Argo CDR is from DSPv2
- if label has value "false" could be a pre-exist cluster CRD

## Description

https://issues.redhat.com/browse/RHOAIENG-4570
based on https://github.com/opendatahub-io/opendatahub-operator/pull/955 

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by ykaliuta** [comment]
Out of curiosity, where this logic comes from? I checked Jira, slack thread, the referenced commit message, but missed it somehow.

**Comment by zdtsw** [comment]
> Out of curiosity, where this logic comes from? I checked Jira, slack thread, the referenced commit message, but missed it somehow.

for all the resource deployed by ODH operator, we set the label and "true"  on these resource, this is  by the "plugin" 
two cases on Argo CRD which might not be from us but done by the user themselves:
- no such label on the Argo CRD
- has such label but it is set to "false"

**Review by ykaliuta — APPROVED** [review]

File: controllers/datasciencecluster/datasciencecluster_controller.go [hunk 1]
```diff
@@ -473,11 +474,13 @@ func (r *DataScienceClusterReconciler) watchDataScienceClusterResources(a client
 var argoWorkflowCRDPredicates = predicate.Funcs{
 	DeleteFunc: func(e event.DeleteEvent) bool {
 		if e.Object.GetName() == datasciencepipelines.ArgoWorkflowCRD {
-			labels := e.Object.GetLabels()
-			if _, ok := labels["app.opendatahub.io/"+datasciencepipelines.ComponentName]; !ok {
-				return true
+			labelList := e.Object.GetLabels()
+			// CRD to be deleted with label "app.opendatahub.io/datasciencepipeline":"true", should not trigger reconcile
+			if value, exist := labelList[labels.ODH.Component(datasciencepipelines.ComponentName)]; value == "true" && exist {
```
Comment (bartoszmajsak) [inline controllers/datasciencecluster/datasciencecluster_controller.go:479]:
```suggestion
			if value, exist := labelList[labels.ODH.Component(datasciencepipelines.ComponentName)]; exist && value == "true" {
```

Comment (zdtsw) [reply controllers/datasciencecluster/datasciencecluster_controller.go:479]: a shortcut :D

File: components/datasciencepipelines/datasciencepipelines.go [hunk 2]
```diff
@@ -159,11 +159,11 @@ func UnmanagedArgoWorkFlowExists(ctx context.Context,
...
 		return fmt.Errorf("failed to get existing Workflow CRD : %w", err)
 	}
-	// Verify if existing workflow is deployed by ODH
-	_, odhLabelExists := workflowCRD.Labels[labels.ODH.Component(ComponentName)]
-	if odhLabelExists {
-		return nil
+	// Verify if existing workflow is deployed by ODH with label
+	odhLabelValue, odhLabelExists := workflowCRD.Labels[labels.ODH.Component(ComponentName)]
+	if odhLabelExists && odhLabelValue != "true" {
+		return fmt.Errorf(" %v CRD already exists but not deployed by this operator. Remove existing Argo workflows or set datasciencepipelines to Removed to proceed ",
```
Comment (etirelli) [inline components/datasciencepipelines/datasciencepipelines.go:165]: maybe use the full path for clarity? spec.components.datasciencepipelines.managedState

**Review by etirelli — APPROVED** [review]
minor suggestion, but other than that LGTM

File: components/datasciencepipelines/datasciencepipelines.go (hunk 2 above)
Comment (zdtsw) [reply components/datasciencepipelines/datasciencepipelines.go:165]: .... it got merged the same time i made this update :rofl:

Comment (zdtsw) [reply components/datasciencepipelines/datasciencepipelines.go:165]: fix in https://github.com/opendatahub-io/opendatahub-operator/pull/970

**Pull request merged**